email_handler.py

```diff
@@ -2608,7 +2658,7 @@ def _handle(self, envelope: Envelope, msg: Message):
             elapsed = time.time() - start
             # Only bounce messages if the return-path passes the spf check. Otherwise black-hole it.
             if return_status[0] == "5":
-                spamd_result = get_spamd_result(msg)
+                spamd_result = get_spamd_result(msg, send_event=False)
```

Comment:
Ah, this one I wanted to discuss with you but totally forgot. Good catch! 
In most of the cases it will already be sent when we do the dmarc policy so we'd be double counting, but there will be some where we bounce before the dmarc policy is checked. I've added the phase (reply, forward or unknown) to the SpamdEvent so we could differentiate. 
Would you still send them even if we double count some events?

Reply:
I wonder if we can call get_spamd_result and store the result that can be used by any method later? get_spamd_result can be called before any email processing starts so we don't rely on dmarc policy check.

Reply:
I've moved all the logic for dmarc and rspamd headers to different files. Now the rspamd info is 'cached' inside the message to prevent parsing the event twice and the inital phase is preserved.